content/blog/tidymodels-2022-q4/index.Rmd

```diff
@@ -0,0 +1,262 @@
+  rowwise() %>%
+  mutate(char = rnorm(1), char = if_else(char > .5, "maroon", "white")) %>%
+  ungroup() %>%
+  rename_with(assign_letter)
+```
+
+```{r}
+str(goofy_data)
+```
+
+Imagine a classification problem on the `goofy_data` where we'd like to predict `class` using the remaining variables as predictors. The selector functions allow us to perform operations on only the predictors with a certain class. For instance, centering and scaling all numeric predictors:
+
+```{r}
+recipe(class ~ ., goofy_data) %>%
+  step_normalize(all_numeric_predictors()) %>%
+  prep()
+```
+
+Or making dummy variables out of each of the nominal predictors:
+
+```{r}
+recipe(class ~ ., goofy_data) %>%
+  step_dummy(all_nominal_predictors()) %>%
+  prep()
+```
+
+Operations like those above have been long-standing functionality in recipes, and are powerful tools for effective modeling. The most recent release of recipes introduced [finer-grain selectors](https://fosstodon.org/@emilhvitfeldt/109315135944110742) for variable types. For instance, we may want to only center and scale the *double* (i.e. real-valued) predictors, excluding the integers. With the new release of recipes, we can easily do so:
+
+```{r}
+recipe(class ~ ., goofy_data) %>%
+  step_normalize(all_double_predictors()) %>%
+  prep()
+```
+
+This is one of a number of new selectors:
+
+-   The `all_nominal()` selector now has finer-grained versions `all_string()`, `all_factor()`, `all_unordered()`, and `all_ordered()`.
+
+-   The `all_numeric()` selector now has finer-grained versions `all_double()`, and `all_integer()`.
+
+-   New `all_logical()`, `all_date()`, and `all_datetime()` selectors.
+
+All new selectors have `*_predictors()` variants. You can read more about recipes 1.0.3 in the [release notes](https://recipes.tidymodels.org/news/index.html#recipes-103).
+
+## Grouped resampling
+
+The most recent release of rsample introduced support for stratification with grouped resampling. Consider the following data set, giving whether a number of `household`s `chops` the `n_melons` melons in their fridge:
+
+```{r}
+#| include: FALSE
+set.seed(1)
+
+household_table <- 
+  tibble(
+    household = 1:100,
+    chops = sample(c(rep("Yes", 90), rep("No", 10)))
+  )
+
+n_melons_table <- 
+  tibble(
+    household = sample(1:100, 5e3, replace = TRUE),
+    n_melons = rexp(5e3, .01)
+  )
+
+melons <- 
+  full_join(household_table, n_melons_table, by = "household") %>%
+  rowwise() %>%
+  mutate(n_melons = if_else(chops == "Yes", n_melons, rpois(1, 2) * n_melons)) %>%
+  ungroup() %>%
+  mutate(n_melons = as.integer(round(n_melons)),
+         household = as.factor(household)) %>%
+  filter(n_melons < 500)
+```
+
+```{r}
+melons
+```
+
+There are `r length(unique(melons$household))` different households in this dataset, each of which either chops their melons or keeps them whole. At different points in time, they have some number of melons in their fridge.
```

Comment:
I think mentioning different time points makes people immediately question whether or not time is relevant. How about we make this multiple members of a household for the repeated measurements? I'm gonna claim that all members of a household adopt the same habit of chopping or not 🤓 i.e. the stratification variable is constant per group.

```suggestion
There are `r length(unique(melons$household))` different households in this dataset. Each member of the household has some number of melons `n_melons` in their fridge. A household, i.e., all its members, either `chops` their melons or keeps them whole.
```

Reply:
Ah, good point.